### PR TITLE
es6 changes to index js file

### PR DESCRIPTION
**Pull request by manoj9788**
upgraded dependencies

Hoping to get this in first and then I will raise another PR with full changes to all the js files under lib and test

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,11 +8,11 @@
...
   },
   "dependencies": {
-    "async": "^1.5.2",
+    "async": "^2.3.0",
     "confit": "^2.0.0",
-    "debug": "^2.1.0",
+    "debug": "^2.6.4",
     "lodash": "^4.12.0",
-    "selenium-webdriver": "^2.53.2",
+    "selenium-webdriver": "2.53.3",
```
Comment (skratchdot) [inline package.json]: Is there a need to pin the version to `2.53.3` or should we make it `^2.53.3`?

File: test/config.js [hunk 2]
```diff
@@ -31,7 +31,7 @@ describe('@config@', function () {
   });
 
   it('should install provided @selenium.version@', function (done) {
-    var ver = '^2.53.1';
+    var ver = '2.53.3';
```
Comment (skratchdot) [inline test/config.js]: Were the tests failing here?  I'm fine w/ this change, but perhaps at some point down the line we need to clean up this test.

**Comment by skratchdot** [comment]
This looks good to me (and was easy to review after adding `?w=1` to the url).  The only other comment I have is that none of the `lib/*.js` files have been converted to es6, but I can take care of that in another PR if you would like.

**Comment by skratchdot** [comment]
Disregard my last comment about the `lib/*.js` files.  I see that you will be submitting a different PR with those changes!  Thanks.

**Comment by grawk** [comment]
You've removed the ^ in both the selenium-webdriver dependency def'n and also the unit test related to installing a specific version. Wondering if you could elaborate on that change? I'm assuming it's got something to do with selenium-webdriver possibly releasing breaking changes in patch versions but not sure?

File: package.json (hunk 1 above)
Comment (manoj9788) [inline package.json]: 2.53.3 is the last version in Selenium 2.0 So makes sense to just 2.53.3 as there will be no more version 2.x.x

File: test/config.js (hunk 2 above)
Comment (manoj9788) [inline test/config.js]: I had all the tests passing

**Comment by manoj9788** [comment]
@grawk Its just that there will be no more 2.x.x releases as we moved on to 3.0.0 I have removed the ^

File: test/config.js (hunk 2 above)
Comment (manoj9788) [inline test/config.js]: > mocha --timeout=30000



  @config@
    ✓ should pass confit object as nemo._config (551ms)
    ✓ should install provided @selenium.version@ (5427ms)
    ✓ should throw an error for invalid @invalid.selenium.version@ (1182ms)
    ✓ should export a Configure method
    ✓ should export a Configure method resolving to a Confit object

  @constructor@
    ✓ should return a promise with @noArguments@
    ✓ should return a promise with @noCallback@ (421ms)
    ✓ should throw an error with @noDriverProps@
    ✓ should launch nemo with @noConfigPath@overrideArg@ (8534ms)
    ✓ should launch nemo with @envConfigPath@noOverrideArg@ (6133ms)
    ✓ should launch nemo with @argConfigPath@noOverrideArg@ (8064ms)
    ✓ should launch nemo with @allArgs@ (4094ms)
    ✓ should return the resolved nemo object when the callback is called (505ms)
    ✓ should resolve when a Confit object is the only parameter (562ms)

  @override@
    ✓ @fromEnv@ over config.json data (472ms)
    ✓ @fromArg@ over config.json data (487ms)
    ✓ @builders@ overrides tgtBrowser abstraction (495ms)

  @plugin@
    ✓ should @priorityRegister@ (467ms)
    ✓ should handle @nonexistPlugin@
    ✓ should handle @failedPluginRegistration@ (476ms)

  @proxy@ 
    ✓ should load problem loading page error (469ms)


  21 passing (38s)

**Comment by manoj9788** [comment]
@scratchdot anything else from my side ? to get this merged

**Comment by grawk** [comment]
I would prefer to have all the es6 changes in one PR as opposed to piecemeal.

**Comment by manoj9788** [comment]
As this was my first PR towards moving into es6, I wanted to see how this first goes, hence just on index.js - Next PR would have all js files under lib and tests.

File: index.js [hunk 3]
```diff
@@ -34,47 +34,48 @@ error.log = console.error.bind(console);
  */
 
 module.exports = function Nemo(_basedir, _configOverride, _cb) {
-  log('Nemo constructor begin');
-  //argument vars
-  var basedir, configOverride, cb, promiz;
-  var nemo = {};
+    log('Nemo constructor begin');
```
Comment (manoj9788) [inline index.js]: done

**Comment by nikulkarni** [comment]
👍  from me. There is one more PR https://github.com/paypal/nemo/pull/161 and part of which attempts similar changes, but I think we should merge this first.

File: index.js [hunk 4]
```diff
@@ -35,13 +35,14 @@ error.log = console.error.bind(console);
...
   //argument vars
-  var basedir, configOverride, cb, promiz;
-  var nemo = {};
+  let basedir, configOverride, cb, promiz;
+  const nemo = {};
 
   //check for confit object as single parameter
   if (arguments.length === 1 && arguments[0].get) {
-    return Setup(arguments[0]);
+      return Setup(arguments[0]);
```
Comment (grawk) [inline index.js]: still 4 space tab

**Comment by grawk** [comment]
still some remaining 4 space tabs

**Comment by manoj9788** [comment]
Anymore changes?

File: package.json (hunk 1 above)
Comment (grawk) [inline package.json]: Is it possible that a bug in 2.53.3 could be found and a patch published with a fix? If so we'd have to change this again for folks to pick up the fix.

Comment (grawk) [inline package.json]: Unless it is certain that no further publishes will ever be made to the selenium-webdriver 2.x family we should leave the carat.

Comment (manoj9788) [inline package.json]: I am pretty sure there won't be new releases in 2.x. But, I understand your concern and adding ^ makes sense. Will change that now.

**Comment by grawk** [comment]
Looks good!

**Pull request merged**